src/xrpld/ledger/detail/ApplyStateTable.cpp

```diff
@@ -260,19 +262,28 @@ ApplyStateTable::apply(
             }
         }
 
-        // add any new modified nodes to the modification set
-        for (auto& mod : newMod)
-            to.rawReplace(mod.second);
+        if (!isDryRun)
+        {
+            // add any new modified nodes to the modification set
+            for (auto const& mod : newMod)
+                to.rawReplace(mod.second);
+        }
 
         sMeta = std::make_shared<Serializer>();
         meta.addRaw(*sMeta, ter, to.txCount());
 
         // VFALCO For diagnostics do we want to show
         //        metadata even when the base view is open?
         JLOG(j.trace()) << "metadata " << meta.getJson(JsonOptions::none);
+
+        metadata = meta;
+    }
+    if (!isDryRun)
```

Comment:
please add new line here

Reply:
Added in 0b818f33287094da7e46370c20269dac49b7a540